components/dashboard/src/projects/project-context.tsx

```diff
@@ -27,50 +27,13 @@ export const ProjectContextProvider: React.FC = ({ children }) => {
     return <ProjectContext.Provider value={ctx}>{children}</ProjectContext.Provider>;
 };
 
-interface ProjectInfo {
-    id: string;
-    name?: string;
-}
-
-export function useProjectInfo(): ProjectInfo | undefined {
-    const projectsRouteMatch = useRouteMatch<{ projectSlug?: string }>("/projects/:projectSlug");
-
-    return useMemo(() => {
-        const projectSlug = projectsRouteMatch?.params.projectSlug;
-        if (!projectSlug) {
-            return undefined;
-        }
-        const result = parseProjectSlug(projectSlug);
-        if (!result) {
-            return undefined;
-        }
-        return result;
-    }, [projectsRouteMatch?.params.projectSlug]);
-}
-
-const pattern: RegExp = /^((.+)-)?([a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12})$/;
-function parseProjectSlug(slug: string): ProjectInfo | undefined {
-    const match = slug.match(pattern);
-
-    if (match) {
-        const name = match[2];
-        const id = match[3];
-        return {
-            name,
-            id,
-        };
-    } else {
-        return undefined;
-    }
-}
-
 export function useCurrentProject(): { project: Project | undefined; loading: boolean } {
     const { project, setProject } = useContext(ProjectContext);
     const [loading, setLoading] = useState(true);
     const user = useCurrentUser();
     const org = useCurrentOrg();
     const orgs = useOrganizations();
-    const projectInfo = useProjectInfo();
+    const projectIdFromRoute = useRouteMatch<{ projectId?: string }>("/projects/:projectId")?.params.projectId;
```

Comment:
```suggestion
    const projectIdFromRoute = useRouteMatch<{ projectId?: string }>("/projects/:projectId")?.params?.projectId;
```

potential read properties of undefined

Reply:
Existing browser sessions might be affected immediately after deployment. I guess it would only impact UX on reload of old URLs.

Reply:
yah, seemed low enough impact to me so avoided handling the old route as well.